# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/painting/nsDisplayList.cpp
# Commit: b04340bad99d
# Full Hash: b04340bad99d70638efda2e7d528e89df77e47c4
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-05-22 09:43:16
# Description:
#   Bug 1617524 - Fix crash in get_relative_transform edge case. r=jnicol
#   
#   Previously, WR would attempt to detect at the start of frame
#   building if the spatial node of any picture cache contained
#   a non-axis-aligned transform, and disable picture caching in
# ==============================================================================

diff -r fdfb16cb82a6 -r b04340bad99d layout/painting/nsDisplayList.cpp
--- a/layout/painting/nsDisplayList.cpp	Thu May 21 18:23:11 2020 +0000
+++ b/layout/painting/nsDisplayList.cpp	Thu May 21 07:42:32 2020 +0000
@@ -6348,6 +6348,9 @@
   if (IsScrollThumbLayer()) {
     params.prim_flags |= wr::PrimitiveFlags::IS_SCROLLBAR_THUMB;
   }
+  if (IsZoomingLayer()) {
+    params.reference_frame_kind = wr::WrReferenceFrameKind::Zoom;
+  }
   StackingContextHelper sc(aSc, GetActiveScrolledRoot(), mFrame, this, aBuilder,
                            params);
 
